lightning/src/onion_message/messenger.rs

```diff
@@ -1473,14 +1473,22 @@ where
 					},
 					#[cfg(async_payments)]
 					ParsedOnionMessageContents::AsyncPayments(AsyncPaymentsMessage::HeldHtlcAvailable(msg)) => {
+						debug_assert!(context.is_none());
```

Comment:
Fixed `peeled_onion_message` and removed this assertion. One thing I'm not sure about is whether to fail `held_htlc_available` messages that have `Some` context, since we expect `None`. I went with ignoring the context for now since we would've failed to decode it if it contained unknown even TLVs.